Bot

HSugioka opened the pull request.

Commit:
update

update not yet

Commit:
add　4点変更

・いいねした人にも１トークン付与
・同一スレッドに1いいねしかできない制限
・集計結果をランキング形式にて記載
・ユーザーに杉岡を追加

LoomClient.jsbのcontractAddressはローカル環境で動作させる為、変更しています。

Commit:
add ランキング

１日のトークン取得数ランキングを毎日一回表示する機能を追加
９時に集計結果を表示した後、そのデータをmysqlへ保存する機能を追加

Commit:
add valid 1いいね/スレッド

いいねを押したスレッドのtimestampをDBへ保存する。もしDBにあればreturn

Commit:
revise

前の状態を再チェック
コメントなどを削除

Commit:
revise

不要なconsole.logを削除
slackへのポストを追加

hnak commented on .vscode/launch.json:
```diff
@@ -0,0 +1,14 @@
+{
```
.DS_Store .vscodeはローカル設定ファイルなのでコミットしないように .gitigore を作成してそこに記載してください。

HSugioka replied:
了解です。

hnak commented on server/src/index.js:
```diff
@@ -35,18 +51,110 @@ rtm.on('reaction_added', (event) => {
+    } catch (err) {
+      throw new Error(err)
+    }
+    for (let i = 0; i < 17; i++) {//17は参加人数が変われば変更の必要あり
```
rows（もしくはusers）に長さを表すメソッドか属性があると思うのでそれを使ってハードコーディングしないようにしてください。

hnak commented on server/src/index.js:
```diff
@@ -3,25 +3,41 @@ const dotenv = require('dotenv');
+        return;
+    }
+    connection.query = util.promisify(connection.query)
+    var rus = await connection.query("SELECT * FROM " + event.user + " where timestamp = '" + event.item.ts + "';")
```
ユーザごとのテーブルを作成するのは運用が大変なので、同じテーブルを使ってください。
カラムにユーザー名を入れた上で、集約関数等を使って必要な情報を取得してください。

hnak reviewed (commented):
レビューコメント入れました。

hnak committed:
Delete .DS_Store

Commit:
delete

vscode削除

Commit:
update add

ハードコーディングを廃止
DBを変更。ユーザーが増えた場合に管理するusersテーブル、同じスレッドへのタイムスタンプを記録するtimestampsテーブルを作成。
loomClientをテスト環境へ変更。
